app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/DataShapeGenerator.java

```diff
@@ -45,20 +45,25 @@
     DataShape createShapeFromResponse(ObjectNode json, T openApiDoc, O operation);
 
     /**
-     * Find parameter that is specified to live in the body.
-     * @param operation holding some parameters.
-     * @return the body parameter.
+     * Find schema that is specified to define the body if any.
+     * @param operation maybe holding a body schema.
+     * @return the body schema.
      */
-    default Optional<OasParameter> findBodyParameter(final OasOperation operation) {
-        if (operation.parameters == null) {
-            return Optional.empty();
-        }
+    default Optional<NameAndSchema> findBodySchema(final O operation) {
+        return Optional.empty();
+    }
 
-        final List<OasParameter> operationParameters = operation.parameters;
+    /**
+     * Combination of schema and name for request body.
+     */
+    class NameAndSchema {
```

Comment:
perhaps?

```suggestion
    class NamedSchema {
```

nitpicking as always...

Reply:
haha, I had exactly this `NamedSchema` first and renamed it into `NameAndSchema` right before the PR.

Reply:
My thoughts were that it is a name and schema because in 2.x name refers to the parameter name and in 3.x to the media type name. Never it is the name of the schema itself. Ah I don't know what's better here

Reply:
Leave it as is, if it ends up bothering us we can rename it... (Naming is hard, remember)